### DISCUSSION TITLE
Adding text line by line from a txt file to different images

### DISCUSSION THREAD

**Discussion by gunisario** [ImageMagick/ImageMagick#2479] [Help]
Hello! Help a newbie. 
ImageMagick 7.0.10-28 Q16 X64 Windows 10. I do this from the command line. 
I need to take in turn each one line from a txt file that has text as a list (line by line) and overlay it on the corresponding image (the list of lines by value matches the order of the images and matches the image name).
For example, the txt file contains 3 lines:
elephant
apple
person
The line with the text "elephant" should overlap with the image with the elephant (elephant.jpg). The line with the text "apple" - to the image with the apple (apple.jpg). I have little experience with PHP, so I am asking for a simple command line code for Windows.
I found a code on the forum that takes text from the file name and overlays it on the image, for this you just need to rename the images - I can, but I am not satisfied with this, I need it from a txt file. So, I found another code:
magick mogrify -path "c:/input/" -pointsize 20 -annotate +50+50 @"c:/output/text.txt" "c:/output/*. jpg"
but this code does not work exactly like that, it adds all lines to all images. As far as I understand, *.jpg means that the text on all images is "elephant apple person", and this is not correct. Is it possible, instead of *.jpg, to specify another txt file in which the list of paths to the image in the same order as the list in text.txt?
Something similar to:
magick mogrify -path "c:/input/" -pointsize 20 -annotate +50+50 @"c:/output/text.txt" **"c:/output/order_images.txt"**

1. How do I fix this so that one line per image, the next line for the next image?
2. Is there a way without PHP?
    I only know one way without PHP:
    convert "c:/output/elephant.jpg" -pointsize 20 -annotate +50+50 "look elephant" "c:/input/elephant_and_text.jpg"
    convert "c:/output/apple.jpg" -pointsize 20 -annotate +50+50 "look apple" "c:/input/apple_and_text.jpg"
    convert "c:/output/person.jpg" -pointsize 20 -annotate +50+50 "look person" "c:/input/person_and_text.jpg"
    

    Hope I was able to explain my thoughts clearly. Thanks in advance for your help.

**Comment by snibgo** [comment]
You have a text file, and want to annotate a series of images, one line of text to each image. How do you know which text line should be annotated on which image?

Your examples have the annotation the same as the filename, eg file apple.jpg needs the annotation "apple". If so, then why do you need the text file? Just loop through the files, annotating from the filename.
